### ISSUE TITLE
Add Deprecation feature to C/C/C Names

### ISSUE THREAD

**Issue by spetillo** [WHOIGit/ooicgsn-roundabout#153]
Initial note to reference conversation in ticket #147

**Comment by ethanandrews** [comment]
@rhmunier this is the deprecation feature  up on rdb-testing you can take a look at

**Comment by rhmunier** [comment]
What I saw causing me to assume Delete instead of Deprecate.  My resolution is 1366x768 which the laptop will not go higher than without an external monitor.

![image](https://user-images.githubusercontent.com/59926007/95379695-b7c15100-08b3-11eb-9ca2-78a9ead667fa.png)

**Comment by spetillo** [comment]
@MarioCarloni In testing on Staging (I did my testing with Configurations for CGINS-ADCPTF-20530), I found a couple of quirks, but this feature generally works as laid out in #147.

Quirks: 

1. If I Remove a CCC Name, I get two warning modals that say the same thing, except the second has a check box so I can silence the warning. Only one modal (with the check box option) should be shown.

2. If I Remove a CCC Name, and that is the only CCC left for an Event, that Event remains with no Names or Values showing. This empty Event should be deleted automatically. (I can at lest delete it manually for now.)

3. If I add a Default to a first Config/Constant Name that exists, then add a second (new) Name in the Part Template, then try to Edit the Defaults to add a default for the second Name, it only shows the Default for the first Name. There is no way to add a Default for the second Name that was added later.

I'll create a bug ticket for each of these and update this comment to tag the tickets shortly. 
**Only item (3) needs to be resolved for v1.5.2 release. Items (1) and (2) are nice to have, but not breaking the functionality.**
